x-pack/plugins/reporting/server/lib/enqueue_job.ts

```diff
@@ -31,24 +41,34 @@ export function enqueueJobFactory(
     context: RequestHandlerContext,
     request: KibanaRequest
   ) {
-    type ScheduleTaskFnType = CreateJobFn<CreateJobBaseParams>;
+    type CreateJobFnType = CreateJobFn<CreateJobBaseParams>;
 
     const exportType = reporting.getExportTypesRegistry().getById(exportTypeId);
 
     if (exportType == null) {
       throw new Error(`Export type ${exportTypeId} does not exist in the registry!`);
     }
 
-    const [scheduleTask, { store }] = await Promise.all([
-      exportType.scheduleTaskFnFactory(reporting, logger) as ScheduleTaskFnType,
+    const [createJob, { store }] = await Promise.all([
+      exportType.createJobFnFactory(reporting, logger) as CreateJobFnType,
       reporting.getPluginStartDeps(),
     ]);
 
-    // add encrytped headers
-    const payload = await scheduleTask(jobParams, context, request);
+    // create a report with encrytped headers in the payload
+    const job = await createJob(jobParams, context, request);
+    const pendingReport = new Report({
+      jobtype: exportType.jobType,
+      created_by: user ? user.username : false,
+      payload: job,
+      meta: {
+        objectType: jobParams.objectType,
+        layout: jobParams.layout?.id,
+      },
+      ...jobSettings,
+    });
 
     // store the pending report, puts it in the Reporting Management UI table
-    const report = await store.addReport(exportType.jobType, user, payload);
```

Comment:
the argument types of addReport changed to take a single report instead of 3 parts of a report

Reply:
Which is why applying `jobSettings` has moved to this file